### ISSUE TITLE
Ivy not working in text mode

### ISSUE THREAD

**Issue by sshbio** [abo-abo/swiper#225]
I installed melpa swiper and counsel from MELPA, and used it in text mode as well as in X, and after your update for your `counsel-yank-pop`, I deleted my .emacs.d and installed everything again (I do not yet know how to deal with updates) then I ran emacs in text mode, and while I make any call to ivy (with a keybinding like `M-x` pointing to `counsel-M-x`) the minibuffer only has an error message:
```
Wrong type argument: symbolp, 0
```

In X it still works pretty well, like it did before.
There is one result for `symbolp` in this repo: [counsel.el](https://github.com/abo-abo/swiper/search?utf8=%E2%9C%93&q=symbolp).
But I also have this error without the package `counsel` installed.

Here are all the info I have thought that could be related to this, but if you want further info, feel free to ask.

`*Packages*`
```
  swiper             20150911.752  installed             Isearch with an overview. Oh, man!
  counsel            20150911.920  installed             Various completion functions using Ivy
```
`M-x version`
```
GNU Emacs 24.5.1 (i586-pc-linux-gnu, GTK+ Version 3.16.4) of 2015-06-29 on x86-csail-01, modified by Debian
```
`$ uname -a`
```
Linux debian 4.1.0-1-686-pae #1 SMP Debian 4.1.3-1 (2015-08-03) i686 GNU/Linux
```
`*Compile-Log*`
```
Leaving directory `/home/josuah/.emacs.d/elpa/counsel-20150911.920'

Compiling file /home/josuah/.emacs.d/elpa/counsel-20150911.920/counsel-pkg.el at Sat Sep 12 12:15:44 2015
Entering directory `/home/josuah/.emacs.d/elpa/counsel-20150911.920/'

Compiling file /home/josuah/.emacs.d/elpa/counsel-20150911.920/counsel.el at Sat Sep 12 12:15:44 2015
counsel.el:1038:1:Warning: defcustom for `counsel-yank-pop-truncate' fails to
    specify containing group
counsel.el:1038:1:Warning: defcustom for `counsel-yank-pop-truncate' fails to
    specify containing group

```
`.emacs`
```elisp
(require 'ivy)
(setq ivy-use-virtual-buffers t)
(global-set-key (kbd "C-c C-r") 'ivy-resume)
(setq ivy-display-style 'fancy)
(ivy-mode 1)

(require 'swiper)
(global-set-key (kbd "C-s") 'swiper)
(global-set-key (kbd "C-r") 'swiper)
(global-set-key (kbd "M-%") 'swiper-query-replace)

(require 'counsel)
(global-set-key (kbd "M-x") 'counsel-M-x)
```

There is no hurry for me, though.

Still, thank you for your update, `counsel-yank-pop` works like a charm!

**Comment by abo-abo** [comment]
You mean it doesn't work for `emacs -nw`? I tried on 24.5.2 GNU/Linux and it still works.

**Comment by sshbio** [comment]
Yes, it does not work for `emacs -nw` nor in TTY. But I got news: I called Swiper with Ctrl-S in an empty scratch buffer, and this time, the Swiper prompt appeared:
```
0    Swiper: 
```
But if I enter a single character in the buffer then call Swiper, the same error occur again.

I will re-install emacs or try the source version and tell if it is works then.

**Comment by abo-abo** [comment]
Thanks, should be fixed now.

**Comment by sshbio** [comment]
Thank you for your fast support !

My computer died an hour after I wrote my last comment. It could be the origin of the issue. The hardware I had was widely unsupported by Linux. I hope the issue was not only due to my configuration or you would be loosing your time...

I tried on my father's iMac, by compiling emacs with and without X support, and without X support, there were an error about fringe mode. The version with x support worked even in a terminal.
A version installed with brew also worked very well.

I also tried on an android tablet with a Debian chroot, and it worked, and also tried with an android equivalent to cygwin (called termux) and it worked again !

It works anywhere emacs works!
